File: Sources/OpenView.swift
import SwiftUI

struct OpenView: View {
    @EnvironmentObject var store: AppStore
    @Environment(\.colorScheme) private var scheme

    var body: some View {
        VStack(spacing: 0) {
            if store.configuration.previewContentEnabled {
                PreviewChip()
                    .padding(.top, 24)
            }

            Spacer()

            RoundedRectangle(cornerRadius: 22, style: .continuous)
                .fill(DT.surface(scheme))
                .frame(width: 96, height: 96)
                .overlay(RoundedRectangle(cornerRadius: 22, style: .continuous)
                    .strokeBorder(DT.hairline(scheme), lineWidth: 0.8))
                .shadow(color: DT.inkLight.opacity(scheme == .dark ? 0 : 0.08), radius: 14, y: 5)
                .overlay {
                    DriveMark()
                        .frame(width: 54, height: 54)
                        .markWiggle()
                }

            Text("DRIVE")
                .font(.system(size: 12, weight: .bold))
                .tracking(4.5)
                .foregroundStyle(DT.ink55(scheme))
                .padding(.top, 20)

            VStack(spacing: 0) {
                Text(store.configuration.previewContentEnabled ? "Talk to your" : "Work with your")
                Text("codebase").foregroundStyle(DT.violetText(scheme))
            }
            .font(.system(size: 34, weight: .heavy))
            .kerning(-1.2)
            .padding(.top, 10)

            Text(store.configuration.previewContentEnabled
                 ? "Watch agents ship while you steer —\nhold to talk, approve every edit."
                 : "Choose a target and connect an approved host.\nNothing runs until you take action.")
                .font(.system(size: 15))
                .foregroundStyle(DT.ink55(scheme))
                .multilineTextAlignment(.center)
                .lineSpacing(3)
                .padding(.top, 14)

            Button {
                if store.hasLiveSession {
                    store.joinCall()
                } else {
                    store.launched = true
                    store.selectedTab = .home
                }
            } label: {
                Text(store.hasLiveSession ? "Watch a live session" : "Open Drive")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundStyle(.white)
                    .frame(maxWidth: .infinity)
                    .frame(height: 52)
                    .background(DT.heroGradient)
                    .clipShape(RoundedRectangle(cornerRadius: DT.rHero, style: .continuous))
                    .shadow(color: DT.violet.opacity(0.35), radius: 14, y: 8)
            }
            .buttonStyle(Pressable())
            .padding(.top, 34)

            if store.configuration.previewContentEnabled {
                Button {
                    store.launched = true
                } label: {
                    Text("Continue with Apple")
                        .font(.system(size: 15, weight: .semibold))
                        .foregroundStyle(DT.ink(scheme))
                        .frame(maxWidth: .infinity)
                        .frame(height: 52)
                        .background(DT.surface(scheme))
                        .clipShape(RoundedRectangle(cornerRadius: DT.rHero, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: DT.rHero, style: .continuous)
                            .strokeBorder(DT.hairline(scheme), lineWidth: 0.8))
                }
                .buttonStyle(Pressable())
                .padding(.top, 12)
            }

            if store.configuration.previewContentEnabled {
                Text("I have an invite link")
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundStyle(DT.violetText(scheme))
                    .padding(.top, 18)
            }

            Spacer()

            if store.configuration.previewContentEnabled {
                HonestyDots()
                    .padding(.bottom, 10)
            } else {
                Text("No edits or file access without your action")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundStyle(DT.ink55(scheme))
                    .padding(.bottom, 10)
            }
        }
        .padding(.horizontal, 36)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background {
            LinearGradient(colors: [scheme == .dark ? DT.pageDark : Color(hex: 0xFBFAFE), DT.page(scheme)],
                           startPoint: .top, endPoint: .bottom)
            .ignoresSafeArea()
        }
    }
}

File: web/src/views/home/OpenView.js
// OpenView.swift — the brand hero and the one violet verb. Preview keeps its
// honesty chrome (PreviewChip, HonestyDots, the Apple stand-in); production
// shows none of that theater and says so in one line.
import { html, useObservable } from "../../ui.js";
import { DriveMark, PreviewChip, HonestyDots } from "../../components.js";
import { ctx } from "./shared.js";

export function OpenView() {
  const s = useObservable(ctx.store);
  const nav = ctx.nav;
  const preview = s.configuration.previewContentEnabled;
  const live = s.hasLiveSession;

  const primary = () => {
    if (live) s.joinCall();
    else { s.launch(); nav.selectTab("home"); }
  };

  return html`<div class="op" data-surface="open">
    ${preview ? html`<${PreviewChip} />` : html`<div style=${{ height: 30 }} />`}
    <div style=${{ flex: 1, minHeight: 24 }} />

    <div class="mark-tile" aria-hidden="true"><${DriveMark} size=${54} wiggle /></div>
    <div class="wordmark">DRIVE</div>
    <h1>${preview ? "Talk to your" : "Work with your"}<br /><span class="verb">codebase</span></h1>
    <p class="lede">${preview
      ? "Watch agents ship while you steer —\nhold to talk, approve every edit."
      : "Choose a target and connect an approved host.\nNothing runs until you take action."}</p>

    <button class="cta pressable" onClick=${primary} aria-label=${live ? "Watch a live session" : "Open Drive"}>
      ${live ? "Watch a live session" : "Open Drive"}
    </button>
    ${preview ? html`<button class="alt pressable" onClick=${() => s.launch()}>Continue with Apple</button>` : null}
    ${preview ? html`<button class="invite pressable" onClick=${() => nav.toast("Invite links open a session directly — none on this device yet.", { icon: "link" })}>I have an invite link</button>` : null}

    <div style=${{ flex: 1, minHeight: 24 }} />
    ${preview
      ? html`<div class="foot" style=${{ paddingTop: 0 }}><${HonestyDots} /></div>`
      : html`<div class="foot" style=${{ paddingTop: 0 }}>No edits or file access without your action</div>`}
  </div>`;
}
